src/Symfony/Component/Security/Core/Authentication/Provider/DaoAuthenticationProvider.php

```diff
@@ -56,7 +57,7 @@ protected function checkAuthentication(UserInterface $user, UsernamePasswordToke
         $currentUser = $token->getUser();
         if ($currentUser instanceof UserInterface) {
             if ($currentUser->getPassword() !== $user->getPassword()) {
-                throw new BadCredentialsException('The credentials were changed from another session.');
+                throw new AccountPasswordChangedException('The credentials were changed from another session.');
```

Comment:
The exception should extend this other type in order to avoid a redirect loop and unset the wrong(old) token from the session.

Both exceptions are correctly handled, checkPreAuth and checkPostAuth in DaoAuthenticationProvider both throw exceptions that extend this base exception class.